### ISSUE TITLE
Peer Dependency Removal

### ISSUE THREAD

**Issue by jimmycann** [mbasso/styled-components-test-utils#7]
**General Information**

- [ ] Bug
- [X] Improvement
- [ ] Feature
- [ ] Other

**Description**

I'll submit a PR for this also, but would like to discuss the possibility of removing peerDependencies from this library.

I'm by no means a master of dependency management, but not sure if there is a need for peerDependencies for this library as it is always installed as a devDependency, therefore all the libraries defined in this packages' devDependencies are available to the upstream project. 

The current implementation hands out npm warnings on install, but even without these peerDependencies in the project, everything continues to function normally.

If you aren't comfortable removing them completely, that's fine too, this could also be an understanding gap on my part. If that's the case, would you mind if I loosen the peerDependency versions for `react`, `react-test-renderer` and `styled-components` instead?

```
npm WARN styled-components-test-utils@0.4.0 requires a peer of chalk@^1.1.3 but none is installed. You must install peer dependencies yourself.
npm WARN styled-components-test-utils@0.4.0 requires a peer of jest-diff@^20.0.3 but none is installed. You must install peer dependencies yourself.
npm WARN styled-components-test-utils@0.4.0 requires a peer of jest-matcher-utils@^20.0.3 but none is installed. You must install peer dependencies yourself.
npm WARN styled-components-test-utils@0.4.0 requires a peer of jest-snapshot@^20.0.3 but none is installed. You must install peer dependencies yourself.
npm WARN styled-components-test-utils@0.4.0 requires a peer of strip-ansi@^3.0.1 but none is installed. You must install peer dependencies yourself.
npm WARN styled-components-test-utils@0.4.0 requires a peer of react-test-renderer@^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN styled-components-test-utils@0.4.0 requires a peer of react@^16.0.0 but none is installed. You must install peer dependencies yourself.
```

Thanks again for your great OSS work!

**Steps to reproduce**

Run `npm install` with this library as a devDependency

**Versions**

- styled-components-test-utils: 0.4.0
- node: 8.10

**Comment by mbasso** [comment]
Hi @yjimk,
yeah, fix these warnings should be great.
We can remove `strip-ansi` without problems, as it is probably an old dependency that I forgot to remove 😖. I think that we can also remove `jest-diff`, `jest-matcher-utils` and `jest-snapshot`, users that want to use this library with jest should already have them, but we shouldn't print a warning if you want to use another test runner.

> If you aren't comfortable removing them completely, that's fine too, this could also be an understanding gap on my part. If that's the case, would you mind if I loosen the peerDependency versions for react, react-test-renderer and styled-components instead?

Some dependencies are required to make this library work, for example, `chalk` is required in all matchers, `react` and `react-test-renderer` are required for example [here](https://github.com/mbasso/styled-components-test-utils/blob/master/src/matchers/toHaveStyleRule.js#L3) and [here](https://github.com/mbasso/styled-components-test-utils/blob/master/src/matchers/toHaveComponent.js#L1). If you haven't them in the `node_modules` folder, installed in the host project (in the `package.json` or as a dependency of another dependency), some functions may not work correctly.
Could you please check if you get warnings because you have different versions installed? For example, we have `chalk@^1.1.3` in our `package.json` but you might have installed `chalk@^2.0.0` that works correctly too.
So, I agree with you, I think that we can relax these dependencies:

```js
{
  "chalk": "^1.0.0 || ^2.0.0",
  "react-test-renderer": "^16.0.0", 
  "react": "^16.0.0", // is v15 compatible with styled-components?
  "styled-components": "^2.0.0 || ^3.0.0" // v2 is also supported
}
```

I think that we should list them to specify which versions can be used with this library and print warnings in the right way. What do you think about that?

Thank you for opening this detailed issue and the PR! I really appreciate your help 😄

**Comment by jimmycann** [comment]
Yeah, very good points there!

I've updated the PR reflecting your comments, also allowing for react@15 and react-test-renderer@15. We're still on 15 for our project and this lib is working a treat.

**Comment by mbasso** [comment]
Awesome! Thank you!
I'll merge and publish a new version as soon as possible

**Comment by mbasso** [comment]
Just published styled-components-test-utils@0.4.1 🎉
Thank you for contributing 😄
